### DISCUSSION TITLE
可以详细解释一下拽拖时候考虑旋转是如何计算的么？

### DISCUSSION THREAD

**Discussion by StringKe** [gausszhou/vue-drag-resize-rotate#6] [General]
可以详细解释一下拽拖时候考虑旋转是如何计算的么？

我在做类似的功能，查看源码后不太清楚旋转之后，方位控制柄的操作逻辑。

**Comment by gausszhou** [comment]
您好，大致思路如下: 
1. 根据旋转角度计算旋转后的矩形坐标（旋转前 => 旋转后）
2. 判断拖拽为顶点还是中点
3. 按下顶点，使其对顶点为不动点，按下中点则选择对面两个顶点中的一个为不动点，其边为临边
4. 鼠标移动时，计算不动点到鼠标位置的向量
5. 顶点模式，计算响亮到两边的投影长度，中点模式只计算向量到临边的投影长度
6. 投影长度为新的矩形的长和宽，根据旋转角度计算旋转前的的矩形坐标（旋转后 => 旋转前）
